Task configure sa: 'Add image pull secrets to a service account' sect… (#21043)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-service-account.md

## Add ImagePullSecrets to a service account

### Create an imagePullSecret

> - Create an imagePullSecret, as described [here](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).

Reviewer: nit:
```suggestion
- Create an imagePullSecret, as described in [Specifying ImagePullSecrets on a Pod](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).
```

    ```shell
    kubectl create secret docker-registry myregistrykey --docker-server=DUMMY_SERVER \
            --docker-username=DUMMY_USERNAME --docker-password=DUMMY_DOCKER_PASSWORD \
            --docker-email=DUMMY_DOCKER_EMAIL
    ```

- Verify it has been created.

>    ```shell script

Reviewer: nit: `shell script` should be just `shell`

   kubectl get secrets myregistrykey
   ```

    The output is similar to this:

    ```
    NAME             TYPE                              DATA    AGE
    myregistrykey    kubernetes.io/.dockerconfigjson   1       1d
    ```

### Add image pull secret to service account

Next, modify the default service account for the namespace to use this secret as an imagePullSecret.

```shell
kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "myregistrykey"}]}'
```

> Alternatively we can use `kubectl edit` or edit manually yaml files as shown below:

Reviewer: This wording would be more idiomatic:
```suggestion
You can instead use `kubectl edit`, or manually edit the YAML manifests as shown below:
```

kubectl replace serviceaccount default -f ./sa.yaml
```

### Verify imagePullSecrets was added to pod spec

> Now, any new pods created in the current namespace and using default service account will have `spec.imagePullSecrets` field added:

Reviewer: How about this wording:
```suggestion
Now, when a new Pod is created in the current namespace and using the default ServiceAccount, the new Pod has its  `spec.imagePullSecrets` field set automatically:
```

````shell
kubectl run nginx --image=nginx --restart=Never

> kubectl get pod nginx -o yaml | grep -A 1 imagePullSecrets

Reviewer: This could also use a [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) query. How about taking that option (rather than `grep`)?

Author: Yes, it is much better!

````

> Output is

Reviewer: nit:
```suggestion
The output is:
```

kubectl replace serviceaccount default -f ./sa.yaml
```

### Verify imagePullSecrets was added to pod spec

Now, when a new Pod is created in the current namespace and using the default ServiceAccount, the new Pod has its  `spec.imagePullSecrets` field set automatically:

````shell
kubectl run nginx --image=nginx --restart=Never
kubectl get pod nginx -o=jsonpath='{.spec.imagePullSecrets[0].name}'
````

The output is:

> ```shell

Reviewer: nit: this is sample output, not shell code
